src/openrct2/interface/Screenshot.cpp

```diff
@@ -220,61 +221,129 @@ static CoordsXY GetEdgeTile(TileCoordsXY mapSize, int32_t rotation, EdgeType edg
     int32_t lower = (visible ? 1 : 0) * COORDS_XY_STEP;
     int32_t upperX = (visible ? mapSize.x - 2 : mapSize.x - 1) * COORDS_XY_STEP;
     int32_t upperY = (visible ? mapSize.y - 2 : mapSize.y - 1) * COORDS_XY_STEP;
-    switch (edgeType)
+    rct_viewport* r = window_get_viewport(window_get_main());
+
+    // check if cut-away view is active, otherwise use the old code instead.
+    if ((r != nullptr && r->flags & VIEWPORT_FLAG_CLIP_VIEW)
+        && (gClipSelectionA.x != 0 || gClipSelectionA.y != 0 || gClipSelectionB.x != MAXIMUM_TILE_START_XY
+            || gClipSelectionB.y != MAXIMUM_TILE_START_XY))
```

Comment:
Wouldn't it be possible to just update lower and upperX/Y to the view clip boundaries instead of having two almost identical switch blocks?

Reply:
Nevermind, there was a single typo.
Commiting very soon

Reply:
Left and right edges are moved to avoid the hard crop at the left and right edge

Reply:
You could see whether or not to use view clipping like this:

```cpp
    auto* const mainWindow = window_get_main();
    const auto* const mainViewport = window_get_viewport(mainWindow);
    const bool useViewClipping = (mainWindow != nullptr && mainViewport->flags & VIEWPORT_FLAG_CLIP_VIEW);
```

Then use that bool to determine the value for the min/max edge tiles, something like this:

```cpp
    const auto lowerX = useViewClipping ? gClipSelectionA.x : 0;
    const auto lowerY = useViewClipping ? gClipSelectionA.y : 0;
    const auto upperX = useViewClipping ? gClipSelectionB.x : ((gMapSize.x - 1) * COORDS_XY_STEP);
    const auto upperY = useViewClipping ? gClipSelectionB.y : ((gMapSize.x - 1) * COORDS_XY_STEP);
```

You can leave the rest almost untouched (only needing to change lower to lowerX and lower Y), and you are nearly there already, without the need to a lot of additional code. You then still need to make sure the returned tiles are included in their entirety, but that's shouldn't be handled differently if view clipping is on or off.

Reply:
`GetTallestVisibleTileTop` also has to take the horizontal clip area into account instead of the full map, as well as the vertical one (not done at all yet). You could loop over just the area like this:

```cpp
    TileCoordsXY startCoords = useViewClipping ? TileCoordsXY(gClipSelectionA) : TileCoordsXY{ 1, 1 };
    TileCoordsXY endCoords = useViewClipping ? TileCoordsXY(gClipSelectionB) : TileCoordsXY{ gMapSize.x - 2, gMapSize.y - 2 };

    int32_t minViewY = 0;
    for (int32_t y = startCoords.y; y <= endCoords.y; y++)
    {
        for (int32_t x = startCoords.x; x <= endCoords.x; x++)
        {
            // ...
```